sdk/monitor/Azure.Monitor.Query/src/LogsQueryClient.cs

```diff
@@ -439,6 +439,7 @@ public virtual Response<LogsQueryResult> QueryResource(ResourceIdentifier resour
             try
             {
                 resource.TrimStart('/');
+                ResourceIdentifier.Parse(resourceId.ToString());
```

Comment:
Should we store the new instance from Parse? Not sure if it does any normalization.

Reply:
Looks like we are calling ToString multiple times.

Reply:
An alternative that would avoid calling ToString again would be to do the following:
```c#
// The Name property will validate that the ResourceIdentifier is valid.
_ = resourceId.Name;
```